Algorithm/QCAlgorithm.Trading.cs

```diff
@@ -46,19 +46,7 @@ public OrderTicket Buy(Symbol symbol, int quantity)
         /// Buy Stock (Alias of Order)
         /// </summary>
         /// <param name="symbol">string Symbol of the asset to trade</param>
-        /// <param name="quantity">double Quantity of the asset to trade</param>
-        /// <seealso cref="Buy(Symbol, int)"/>
-        public OrderTicket Buy(Symbol symbol, double quantity)
```

Comment:
I've included overloads for double decimal and int in the latest commit. I've maintained float support but have not added additional methods. Have also added a dedicated regression test.